clusterloader2/pkg/measurement/common/network/performance-measurement-helper.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package network
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func (npm *networkPerformanceMeasurement) validate(config *measurement.Config) error {
+	var ratio, protocol string
+	var duration int
+	var err error
+
+	if duration, err = util.GetInt(config.Params, "duration"); err != nil {
+		return err
+	}
+
+	if ratio, err = util.GetString(config.Params, "ratio"); err != nil {
+		return err
+	}
+
+	if !checkRatioSeparator(ratio) {
+		return fmt.Errorf("ratio separator : missing")
+	}
+
+	podReplicas, err := util.GetInt(config.Params, "podReplicas")
+	if err != nil {
+		return err
+	}
+	if podReplicas <= 0 {
+		return fmt.Errorf("invalid value of podReplicas , should be > 0")
+	}
+
+	if ok, err := validateClientServerPodNum(ratio); !ok {
+		return err
+	}
+
+	if protocol, err = util.GetString(config.Params, "protocol"); err != nil {
+		return err
+	}
+
+	if protocol != protocolTCP && protocol != protocolUDP && protocol != protocolHTTP {
+		return fmt.Errorf("invalid protocol , supported ones are TCP,UDP,HTTP")
+	}
+
+	npm.testDuration = duration
+	npm.podRatio = ratio
+	npm.protocol = protocol
+	npm.podReplicas = podReplicas
+	return nil
+}
+
+func checkRatioSeparator(ratio string) bool {
+	if !strings.Contains(ratio, ratioSeparator) {
+		return false
+	}
+	return true
+}
+
+func validateClientServerPodNum(ratio string) (bool, error) {
+	var podNumber []string
+	podNumber = strings.Split(ratio, ratioSeparator)
+	clientPodNum, err := strconv.Atoi(podNumber[0])
+	if err != nil {
+		return false, err
+	}
+
+	serverPodNum, err := strconv.Atoi(podNumber[1])
+	if err != nil {
+		return false, err
+	}
+
+	if clientPodNum <= 0 || serverPodNum <= 0 {
+		return false, fmt.Errorf("invalid pod numbers")
+	}
+	return true, nil
+}
+
+func deriveClientServerPodNum(ratio string) (int, int, string) {
+	var podNumber []string
+	var clientPodNum, serverPodNum int
+
+	podNumber = strings.Split(ratio, ratioSeparator)
+	clientPodNum, _ = strconv.Atoi(podNumber[0])
+	serverPodNum, _ = strconv.Atoi(podNumber[1])
```

Comment:
these things always scare me, a wrong integer gives a 0 IIRC, show we check the error?

Reply:
Yes this code is modified now.